packages/identity-sdk/src/viem-sdk.ts

```diff
@@ -0,0 +1,215 @@
+import {
+  Address,
+  PublicClient,
+  WalletClient,
+  parseAbi,
+  SimulateContractParameters,
+  WalletActions,
+} from "viem";
+import { waitForTransactionReceipt } from "viem/actions";
+import { z } from "zod";
+import { compressToEncodedURIComponent } from "lz-string";
+
+import { Envs, FV_IDENTIFIER_MSG2 } from "./constants";
+
+const identityV2ABI = parseAbi([
+  "function addWhitelisted(address account)",
+  "function removeWhitelisted(address account)",
+  "function isWhitelisted(address account) view returns (bool)",
+  "function lastAuthenticated(address account) view returns (uint256)",
+  "function authenticationPeriod() view returns (uint256)",
+]);
+
+const addressSchema = z
+  .string()
+  .regex(/^0x[a-fA-F0-9]{40}$/, "Invalid Ethereum address");
+
+interface IdentityContract {
+  publicClient: PublicClient;
+  contractAddress: Address;
+  abi: any;
+}
+
+interface IdentityExpiryData {
+  lastAuthenticated: bigint;
+  authPeriod: bigint;
+}
+
+export const initializeIdentityContract = (
+  publicClient: PublicClient,
+  contractAddress: Address,
+): IdentityContract => ({
+  publicClient,
+  contractAddress,
+  abi: identityV2ABI,
+});
+
+export const submitAndWait = async (
+  params: SimulateContractParameters,
+  publicClient: PublicClient,
+  walletClient: WalletClient & WalletActions,
+  onHash?: (hash: `0x${string}`) => void,
+) => {
+  try {
+    const [account] = await walletClient.getAddresses();
+    if (!account) throw new Error("No active wallet address found.");
+
+    const { request } = await publicClient.simulateContract({
+      account,
+      ...params,
+    });
+
+    const hash = await walletClient.writeContract(request);
+    onHash?.(hash);
+
+    const receipt = await waitForTransactionReceipt(publicClient, { hash });
+
+    if (!receipt) {
+      throw new Error(`Transaction receipt not found for hash ${hash}`);
+    }
+
+    if (receipt.status !== "success") {
+      throw new Error(`Transaction failed with status ${receipt.status}`);
+    }
+
+    return receipt;
+  } catch (error: any) {
+    console.error("submitAndWait Error:", error);
+    throw new Error(`Failed to submit transaction: ${error.message}`);
+  }
+};
+
+export const isWhitelisted = async (
+  contract: IdentityContract,
+  accountToCheck: Address,
+): Promise<boolean> => {
+  const validation = addressSchema.safeParse(accountToCheck);
+  if (!validation.success) {
+    throw new Error(
+      `isWhitelisted Error: ${validation.error.issues.map((issue) => issue.message).join(", ")}`,
+    );
+  }
```

Comment:
why do we need this? if address is incorrect an error will be thrown else where below

Reply:
I would say not having to make an rpc call to wait for it to be rejected there. instead, catch it early

Reply:
there's no rpc call invovled. if address is incorrect it will fail before, in any case, i dont know why you suddenly started adding all these type checks, imho they are redundant, their are the dev responsibility to handle and parse

Reply:
The 'why of a sudden'. its AI following some principles and rules for safety. for me, it did not seem to be in the way of anything why I left it as that.

Removed no upon your review